neuralpredictors/layers/readouts.py

```diff
@@ -988,11 +1003,24 @@ def forward(self, x, sample=None, shift=None, out_idx=None):
         if shift is not None:
             grid = grid + shift[:, None, None, :]
 
-        y = F.grid_sample(x, grid, align_corners=self.align_corners)
-        y = (y.squeeze(-1) * feat).sum(1).view(N, outdims)
+        if not multiplex:
+            y = F.grid_sample(x, grid, align_corners=self.align_corners)
+            y = (y.squeeze(-1) * feat).sum(1).view(N, outdims)
+        else:
+            y = torch.einsum("ncwh,uco->nwho", x, feat)
 
         if self.bias is not None:
             y = y + bias
+
+        if multiplex:
+            if crop_edge_px:
+                y = y[:, crop_edge_px:-crop_edge_px, crop_edge_px:-crop_edge_px, :]
+            if collapse:
+                y = y.reshape(N, -1)
+            if not collapse:
```

Comment:
```suggestion
           else:
```
Is there a reason for splitting it into two checks?